addons/mail/models/mail_thread.py

```diff
@@ -943,13 +943,8 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
         #        As all MTA does not respect this RFC (googlemail is one of them),
         #       we also need to verify if the message come from "mailer-daemon"
         #    If not a bounce: reset bounce information
-        if bounce_alias and any(email.startswith(bounce_alias) for email in email_to_localparts):
-            bounce_re = re.compile("%s\+(\d+)-?([\w.]+)?-?(\d+)?" % re.escape(bounce_alias), re.UNICODE)
-            bounce_match = bounce_re.search(email_to)
-            if bounce_match:
-                self._routing_handle_bounce(message, message_dict)
-                return []
-        if message.get_content_type() == 'multipart/report' or email_from_localpart == 'mailer-daemon':
+        bounce_match = bounce_alias and any(email.startswith(bounce_alias) for email in email_to_localparts)
```

Comment:
I would keep the old regex-based computation (for compatibility) and use a strict equal when using the ICP, to avoid having "bounce.robert" being considered as a write to bounce alias due to startswitch. Robert Bounce could be sad if all its emails are considered as bounces ...